providers/flagd/src/main/java/dev/openfeature/contrib/providers/flagd/resolver/grpc/cache/Cache.java

```diff
@@ -1,41 +1,42 @@
 package dev.openfeature.contrib.providers.flagd.resolver.grpc.cache;
 
 import dev.openfeature.sdk.ProviderEvaluation;
+import lombok.Getter;
+import lombok.extern.slf4j.Slf4j;
 import org.apache.commons.collections4.map.LRUMap;
 
 import java.util.Collections;
 import java.util.Map;
 
+import static dev.openfeature.contrib.providers.flagd.resolver.grpc.cache.CacheType.DISABLED;
+import static dev.openfeature.contrib.providers.flagd.resolver.grpc.cache.CacheType.LRU;
+
 /**
  * Exposes caching mechanism for flag evaluations.
  */
+@Slf4j
 public class Cache {
-    private Map<String,ProviderEvaluation<? extends Object>> store;
-    private Boolean enabled = false;
+    private Map<String, ProviderEvaluation<? extends Object>> store;
+
+    @Getter
+    private final Boolean enabled;
 
     /**
      * Initialize the cache.
-     * @param type of cache.
+     *
+     * @param forType      type of the cache.
      * @param maxCacheSize max amount of element to keep.
      */
-    public Cache(CacheType type, int maxCacheSize) {
-        if (type == null) {
-            return;
-        }
-
-        switch (type) {
-            case DISABLED:
-                return;
-            case LRU:
-            default:
-                this.store = Collections.synchronizedMap(new LRUMap<>(maxCacheSize));
+    public Cache(final String forType, int maxCacheSize) {
+        if (DISABLED.getValue().equals(forType)) {
+            enabled = false;
+        } else if (LRU.getValue().equals(forType)) {
+            enabled = true;
+            this.store = Collections.synchronizedMap(new LRUMap<>(maxCacheSize));
+        } else {
+            enabled = false;
+            log.info(String.format("Unsupported cache type %s, continuing without cache", forType));
```

Comment:
```suggestion
            log.warn(String.format("Unsupported cache type %s, continuing without cache", forType));
```

This feels like a warn to me.

Reply:
yeah, I agree on this